aim/storage/rockscontainer.py

```diff
@@ -487,3 +488,32 @@ def prev_key_value(
         key, value = it.get()
 
         return key, value
+
+    def optimize_db_for_read(self):
+        """
+        This function will try to open rocksdb db in write mode and force WAL files recovery. Once done the underlying
+        db will contain .sst files only which will significantly reduce further open and read operations. Further
+        optimizations can be done by running compactions but this is a costly operation to be performed online.
+        """
+        assert self.read_only
+
+        def non_empty_wal():
+            for wal_path in self.path.glob('*.log'):
+                if os.path.getsize(wal_path) > 0:
+                    return True
+            return False
+
+        if non_empty_wal():
+            lock_path = self.prepare_lock_path()
+
+            with FileLock(str(lock_path), timeout=10):
```

Comment:
@alberttorosyan jan won't this throw an error, if lock couldn't be acquired inside timeout (this would be the case if the run is still actually in progress)? The read operation would fail in this case

Reply:
good point. will add exception handling